addons/mail/models/mail_channel.py

```diff
@@ -61,17 +61,20 @@ class Channel(models.Model):
 
     MAX_BOUNCE_LIMIT = 10
 
-    def _get_default_image(self):
-        image_path = modules.get_module_resource('mail', 'static/src/img', 'groupdefault.png')
-        return base64.b64encode(open(image_path, 'rb').read())
-
     @api.model
     def default_get(self, fields):
         res = super(Channel, self).default_get(fields)
         if not res.get('alias_contact') and (not fields or 'alias_contact' in fields):
             res['alias_contact'] = 'everyone' if res.get('public', 'private') == 'public' else 'followers'
         return res
 
+    def _default_channel_last_seen_partner_ids(self):
+        return [(0, 0, {"partner_id": self.env.user.partner_id.id})]
```

Comment:
This change created a new problem: website visitors opening the livechat now cause the "Public" partner to be included in the livechat channel, because `self.env.user` is the Public user here. In terms of permissions, Public is never supposed to be a member of livechat channel.

This has other undesired consequences, for example new leads created with the `/lead` livechat command will now all be linked to the Public partner, creating an accumulated mess as contact data overrides previous contact data, all the Public partner data, etc.

This should probably be guarded with a test on `active`, to skip "website public users":

```py
def _default_channel_last_seen_partner_ids(self):
    if self.env.user.active:
        return [(0, 0, {"partner_id": self.env.user.partner_id.id})]
    return []
```

Reply:
@odony It's actually necessary for us to know the "public partner" is part of the channel. If you remove that, the rest of the livechat client will be bugged in many ways.

We need to be able to distinguish livechat with guest from livechat with logged in users and from livechat with guest on which a second operator has been invited. Notably without having the public partner, the last 2 cases would have the same member list, but represent two completely different situations.

This mess is the main reason why for v15 we want to automatically create actual partners when having a livechat conversations.

If there is an issue with `/lead` command or other parts of the code because of that, it needs to be handled case by case.

Reply:
Hello guys, I have read what you said and I understand your arguments @seb-odoo .
So, I am wondering if this code still makes sense, as it was done to filter out the public user which is the only case where the environment's user is not active.

https://github.com/odoo/odoo/blob/78e6ce3c3fe565c488b956450ed875629cdd2de0/addons/im_livechat/models/im_livechat_channel.py#L114

It is confusing to have a code that explicitly filter out the public user, then have a hidden function that takes advantage of the `default` behavior of a field to have the filtered out user added.

What do you think?

Reply:
Maybe this line is not useful anymore, I'm not sure like that. But it's very much possible that condition was introduced before the change of default indeed.
Is that line causing an issue? I don't understand why we worry about it.